starter/src/test/java/com/starter/fullstack/dao/InventoryDAOTest.java

```diff
@@ -56,7 +56,20 @@ public void createTest() {
     Assert.assertTrue(inventory1.getName() == NAME);
     Assert.assertTrue(inventory1.getProductType() == PRODUCT_TYPE);
   }
-
+  /**
+   * Test delete method
+   */
+  @Test
+  public void deleteTest() {
+    Inventory inventory = new Inventory();
+    inventory.setName(NAME);
+    inventory.setId(ID);
+    inventory.setProductType(PRODUCT_TYPE);
+    inventory = this.mongoTemplate.save(inventory);
+    String id = inventory.getId();
+    //isPresent confirms that the Inventory still exists but is only deleted from the inventoryDAO
```

Comment:
This is going to seem nit picky, but in this case an important distinction.
`isPresent()` doesn't say anything about the fact that the item exists specifically, it tells you whether or not a value, an instance of Inventory, is contained within the Optional container.
The `findAndRemove` method from mongo either returns the item it found and deleted, or returns null to indicate that nothing was returned (which we have wrapped in the Optional container)

Your best bet is going to be to either update the comment with this new understanding or to just leave the comment off as it's unfortunately incorrect. Personally I'd recommend the first option. (for your future self if no one else)